frontend/catalyst/pennylane_extensions.py

```diff
@@ -1169,8 +1189,8 @@ def qctrl_distribute(
         else:
             ops2.append(
                 Controlled(
-                    type(op)(*op.parameters, wires=op.wires),
-                    control_wires=qml.wires.Wires(control_wires),
+                    op,
```

Comment:
By linking this operation by ref we become vulnerable by later changes of parameters or wires of this operation (made by other transformations). I would suggest to keep the copying.

Reply:
That was going to be my question to you, but the line of code you had was causing problems (not every class appearing here has a wires keyword argument). Maybe we could use the copy method instead?

Reply:
So it looks like the operator class does implement copy and deepcopy methods: https://github.com/PennyLaneAI/pennylane/blob/89a7e775a7c8e444ee62fb411946bbdb15ecaa86/pennylane/operation.py#L691

But I'm also curious why you think a class with fields would not be handled well by the generic copy function? For one, a class without fields (a mere collection of methods) would surely not be meaningful to copy to begin with.
As far as I know, the standard copy function would just create new instance and (shallow) copy each field to the new instance.

The other thing I'm wondering is why an instantiation with `parameters` and `wires` arguments only (the previous implementation) would produce a better copy than cloning an objective instance by copying all its fields. Unless the object being copied has a strict contract that those two arguments fully specify an object instance (I'm worried this might not be the case, already we know `Pow` doesn't even have a wires argument).

Reply:
> So it looks like the operator class does implement copy and deepcopy methods

I did not know this, thank you! In this case, I think we must use these copy methods in transformations. My original approach had this goal as well. I think it would keep the semantics of the transformations simple. With links, we allow newer transformations to affect older ones after they are completed, which is hard to describe, formalize and test.

Reply:
Used the copy method here: [4507cde](https://github.com/PennyLaneAI/catalyst/pull/408/commits/4507cde3285a9e02a2c570d0e9f03ca1487d3123)